lib/livebook/hubs/provider.ex

```diff
@@ -0,0 +1,27 @@
+defprotocol Livebook.Hubs.Provider do
+  @moduledoc false
+
+  @type t :: %{
+          required(:__struct__) => module(),
+          required(:id) => String.t(),
+          optional(any()) => any()
+        }
+
+  @doc """
+  Normalize given struct to `Metadata` struct.
```

Comment:
It's not crucial because we don't publish docs, but note that in order for ex_doc to generate links, we need a fully qualified module name :)